src/main/java/io/github/zlika/reproducible/FileConsumer.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.zlika.reproducible;
+
+import java.io.File;
+import org.apache.maven.plugin.MojoExecutionException;
+
+/**
+ * A consumer implementing the logic required to strip a file.
+ * @author Umberto Nicoletti (umberto.nicoletti@gmail.com)
+ * @version $Id$
+ */
+interface FileConsumer
```

Comment:
Creating a new interface just because the "strip" method of interface Stripper throws IOException instead of MojoExecutionException seems overkill. Moreover, if you look at the implementation classes, there are in fact throwing an IOException that is catched to be replaced by a MojoExecutionException!
It seems that this new interface only add complexity for no obvious benefit.

Reply:
@Zlika Thanks for the feedback and I see your point :-)

This refactoring was triggered by checkstyle complaining that the `strip()` method was too long and indeed it was too long and contained duplicated code. Initially I wanted to use a `Consumer` lambda, but then I could not throw any checked exception, so I took this slightly longer path.
The benefit of the interface is that it removes a some more duplicated code (not a lot, tbh). :-)

I could drop the interface, but I'd keep the 2 implementations and adapt `strip` and `process` accordingly. Wdyt?